app/lib/katello/util/thread_session.rb

```diff
@@ -54,9 +54,16 @@ def self.current=(o)
             end
 
             def self.pulp_config(user_remote_id, &_block)
-              uri = URI.parse(SETTINGS[:katello][:pulp][:url])
+              Katello.pulp_server = runcible_instance(SETTINGS[:katello][:pulp][:url], user_remote_id)
+              yield if block_given?
```

Comment:
:+1: TBH not sure why pulp server needs to be in a thread? anyway /cc @shlomizadok as you had similar issue